app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,22 @@
+class SessionsController < ApplicationController
+  skip_before_action :require_user
+  before_action :redirect_when_logged_in, only: :new
+
+  def create
+    user = User.find_by(email: params[:email])
+    if user && user.authenticate(params[:password])
+      session[:current_user_id] = user.id
+      flash[:success] = "You have successfully logged in."
+      redirect_to root_path
+    else
+      flash[:danger] = "Please check your username and password and try again."
```

Comment:
I would recommend `flash.now` when you render. Otherwise the flash pops on next request and we know that render is not a redirect. flash is built to pop on next GET request..